app/models/asset.rb

```diff
@@ -94,9 +95,16 @@ def dimensions_for(style)
   end
 
   def reprocess_for(style)
+    return if processing?
+
     if file.styles.has_key?(style) && ! file_dimensions.has_key?(style.to_s)
-      file.reprocess!(style)
-      save
+      begin
+        set(:processing, true)
```

Comment:
Noting for posterity:

We had an exciting conversation in Slack wherein @bensymonds suggested the following as a safe alternative to `return if processing?`:

```rb
return unless Asset.where(id: id, processing: false).set(processing: true)["updatedExisting"]
```